tap_sftp/client.py

```diff
@@ -108,6 +109,19 @@ def get_files_by_prefix(self, prefix):
                 if is_empty(file_attr):
                     continue
 
+                # skip gzip file if it is empty
+                if file_attr.filename.endswith('.gz'):
+                    with self.sftp.open(prefix + '/' + file_attr.filename, "rb") as file:
+                        with gzip.GzipFile(fileobj=file, mode='rb') as gzip_file:
+                            try:
+                                data = gzip_file.read()
+                            except OSError:
```

Comment:
@harshpatel4crest Please check a scenario where a gzipped file is not given permission to read to a non-root user. What will be the tap behavior in this case?

Reply:
@harshpatel4crest Also, check the scenario where a file is gzipped with "--no-name"

Reply:
Done